stackit/internal/services/objectstorage/bucket/resource.go

```diff
@@ -0,0 +1,295 @@
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "ObjectStorage bucket client configured")
+}
+
+// Schema defines the schema for the resource.
+func (r *bucketResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":                     "ObjectStorage bucket resource schema.",
+		"id":                       "Terraform's internal resource identifier. It is structured as \"`project_id`,`bucket_name`\".",
+		"bucket_name":              "The bucket name. Has to be DNS conform.",
+		"project_id":               "STACKIT Project ID to which the bucket is associated.",
+		"url_path_style":           "URL in path style.",
+		"url_virtual_hosted_style": "URL in virtual hosted style.",
+	}
+
+	resp.Schema = schema.Schema{
+		Description: descriptions["main"],
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: descriptions["id"],
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"bucket_name": schema.StringAttribute{
+				Description: descriptions["bucket_name"],
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.NoSeparator(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Description: descriptions["project_id"],
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"url_path_style": schema.StringAttribute{
+				Computed: true,
+			},
+			"url_virtual_hosted_style": schema.StringAttribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *bucketResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) { // nolint:gocritic // function signature required by Terraform
+	var model Model
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	projectId := model.ProjectId.ValueString()
+	bucketName := model.BucketName.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "bucket_name", bucketName)
+
+	// Create new recordset
+	_, err := r.client.CreateBucket(ctx, projectId, bucketName).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating bucket", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+
+	wr, err := objectstorage.CreateBucketWaitHandler(ctx, r.client, projectId, bucketName).SetTimeout(1 * time.Minute).WaitWithContext(ctx)
```

Comment:
Why just 1 minute? Even if it is really fast, why should we have such a small timeout?

Reply:
From my tests, creation was always under 5s and deletion was under 15s. Set the timeout as 1 minute because it seems like, if it would take that long, something is definitely wrong.

Reply:
Ok lets start with this and then check again if there are a lot of timeouts

Reply:
It's also used by other resources as well